src/coordinator/cluster/m3db/async_client.go

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
```

Comment:
Would it make sense to put this next to `asyncsession` in `src/coordinator/stores/m3db/` instead? Also, can you rename one of the files to keep naming consistent?

Reply:
Not really because it's "stores" which implies time series storages.  This is why I created the `cluster` package which describes clustering backends.

Reply:
I'll rename the asyncsession file though, missed that - good point.